src/Functions/formatDateTime.cpp

```diff
@@ -1032,6 +1032,12 @@ class FunctionFormatDateTimeImpl : public IFunction
                         out_template += "00";
                         break;
 
+                    // Full month [January-December]
+                    case 'M':
+                        instructions.emplace_back(&Action<T>::mysqlMonthOfYearTextLong);
+                        out_template += "January";
```

Comment:
If I read the code correctly, the output is created in two steps: first, it is populated by `out_template` and second, it is overwritten by the actual (computed) value. This works all fine if the modifier produces a fixed-size output (e.g. `00/00/00` or `Jan`) but one must be careful with variable-size outputs (e.g. the month names). E.g. `February` has more characters than `January` (l. 1038). Would you like to check that everything works (e.g. no writing behind the allocated space) if the printed month isn't `January`? (e.g. try `February` and `May`)

Reply:
Ah, just see that Alexey already made a comment along the same lines :)

Reply:
Hey rschu1ze! You're right on the money, well spotted :) I have already written my concerns in the changelog description at the top (under TODO heading). You'll see I have been running into this exact issue. I have spent some time today giving the code a more thorough review and it's a little bit fiddly to navigate about a solution for the mySQL method but I have come up with a plan I'm going to try this evening.

I am going to try modifying the `out_template` string with the specific month rather than using  default template i.e `January`, ideally using some kind of simple callback function.

Reply:
@rschu1ze So I've really struggled with actually getting a nice implementation working for dynamic string sizes. The current setup doesn't allow for dynamic string sizes by design and I'm lacking the skills / averse to making a large change I'm ashamed to say.

It feels like it should be simple, we just need to pass the full-month string size to `out_template` in the `parseMySQLFormat` method, but we don't have the arguments available to us at the time to do so.

I think I'm going to go with the route of padding out values with spaces if they are too small (e.g `May      `). I'm not totally satisfied with this approach but allows me to get this change out the door and helping out with more work that the team needs done. I'm always open to suggestions / ideas if you have any :)

Reply:
Padding approach added as second commit.